src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoOfT.cs

```diff
@@ -234,7 +234,7 @@ public override bool ReadJsonAndSetMember(object obj, ref ReadStack state, ref U
             else
             {
                 success = true;
-                if (!isNullToken || !IgnoreDefaultValuesOnRead || !Converter.CanBeNull)
+                if (!isNullToken || !IgnoreDefaultValuesOnRead || !Converter.CanBeNull || state.IsContinuation)
```

Comment:
In the [`if` statement above here](https://github.com/dotnet/runtime/blob/4b3e724919da9eacc1e63a7bbb9c5b4b08b10acd/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoOfT.cs#L223-L233)  please add an Assert:
```cs
            else if (Converter.CanUseDirectReadOrWrite && state.Current.NumberHandling == null)
            {
                Debug.Assert(!state.IsContinuation);
```

The Assert will be true since `CanUseDirectReadOrWrite` is false when using a Stream.

Reply:
Added assertions here and in `ReadJsonAsObject`.